istioctl/cmd/istioctl/convert/cmd.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convert
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	"github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/pilot/pkg/config/kube/crd"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+)
+
+var (
+	inFilenames []string
+	outFilename string
+)
+
+// Command for converting v1alpha1 configs to v1alpha3
+func Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "convert",
```

Comment:
`convert` is a pretty general term for a command that only converts a subset of Istio config from one alpha version to a newer alpha version. Maybe something like `convert-networking`? Given that this should hopefully be short lived, relative to lifespan of Istio, we should also consider introduce an `experimental` or `alpha` subcommand group which has relaxed backwards compat. guarantees. This is analogous to `cloud alpha`  e.g.

`istioctl alpha convert-networking-apis -f <in.yaml> -o <out.yaml>`